### PR TITLE
WebGPU Update

### PR DESCRIPTION
**Pull request by sebavan**

### CODE REVIEW

File: packages/dev/core/src/Engines/WebGPU/webgpuShaderProcessorsWGSL.ts [hunk 1]
```diff
@@ -494,4 +508,11 @@ export class WebGPUShaderProcessorWGSL extends WebGPUShaderProcessor {
 
         return code;
     }
+
+    private _processStridedUniformArrays(code: string): string {
+        for (const uniformArrayName of this._stridedUniformArrays) {
+            code = code.replace(new RegExp(`${uniformArrayName}\\[(.*)\\]`, "g"), `${uniformArrayName}[$1].el`);
```
Comment (Popov72) [inline packages/dev/core/src/Engines/WebGPU/webgpuShaderProcessorsWGSL.ts:514]: I think syntax like `myarray [23]` is allowed in wgsl, so a `\s*` should be added in the regexp?

Something like: `${uniformArrayName}\\s*\\[(.*)\\]`

**Review by Popov72 — APPROVED** [review]
LGTM!

What a mess we have to do because they removed `@stride`...

File: packages/dev/core/src/Engines/WebGPU/webgpuShaderProcessorsWGSL.ts [hunk 2]
```diff
@@ -365,23 +369,33 @@ export class WebGPUShaderProcessorWGSL extends WebGPUShaderProcessor {
...
         let ubo = `struct ${name} {\n`;
         for (const leftOverUniform of this._webgpuProcessingContext.leftOverUniforms) {
             const type = leftOverUniform.type.replace(/^(.*?)(<.*>)?$/, "$1");
             const size = WebGPUShaderProcessor.UniformSizes[type];
 
             if (leftOverUniform.length > 0) {
                 if (size <= 2) {
-                    ubo += ` @align(16) ${leftOverUniform.name} : @stride(16) array<${leftOverUniform.type}, ${leftOverUniform.length}>;\n`;
+                    const stridedArrayType = `${name}_${stridedArraysLength}_strided_arr`;
+                    stridedArrays += `struct ${name}_${stridedArraysLength}_strided_arr {
```
[the remark this answers is not in the document]
Comment (sebavan) [reply packages/dev/core/src/Engines/WebGPU/webgpuShaderProcessorsWGSL.ts]: Good catch

File: packages/dev/core/src/Engines/WebGPU/webgpuShaderProcessorsWGSL.ts (hunk 1 above)
Comment (sebavan) [reply packages/dev/core/src/Engines/WebGPU/webgpuShaderProcessorsWGSL.ts]: yup

**Comment by sebavan** [comment]
> LGTM!
> 
> What a mess we have to do because they removed `@stride`...

Yes, this is insane... and what they now do in tint.

**Review by deltakosh — APPROVED** [review]

**Pull request merged**